### PR TITLE
Added HanaOnAzure spec

### PR DESCRIPTION
**Pull request by PakDLiu**
SAP HANA is a new resource the AzLinux team is adding. This resource is going to be served from the HANA resource provider.
This change will add the new resource for other service to consume (currently mainly for Azure CLI).

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by olydis** [comment]
@simongdavies @ravbhatnagar new spec

**Comment by olydis** [comment]
@PakDLiu while waiting for ARM review, note that there are model validation errors, i.e. your examples don't entirely match the Swagger definition: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/313187258#L706 Please fix either definition or example. 🙂

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (ravbhatnagar) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: enum?

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (ravbhatnagar) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: Are these values a known set?

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (ravbhatnagar) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: Add a default response to capture all the error responses. The model must comply with error contract as per RPC. See the example in the swagger for batch RP.

Comment (ravbhatnagar) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: For all APIs this must be added.

**Review by ravbhatnagar — COMMENTED** [review]
Overall looks good. Some minor comments. SDK team- consider sign off from ARM team based on the response to the minor issues noted.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: Hey, I am not sure what you mean. What is RPC? and where can I find exmaple in the swagger for batch RP?

Are you saying I need something like
```
"204": {
            "description": "No Content"
          }
```
for default response?

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: Yes, I will find out all the available sizes and make this an enum

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (ravbhatnagar) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: here is an example - https://github.com/Azure/azure-rest-api-specs/blob/current/specification/batch/resource-manager/Microsoft.Batch/2017-09-01/BatchManagement.json#L101

**Comment by ravbhatnagar** [comment]
@olydis - Looks good to merge from ARM side after the default response has been added.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: done

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/hanaonazure.json]: done

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (olydis) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: our naming convention here is `Operations_List`, since the client that this operation is on will already contain the "subject", i.e. "Hana" in this case

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (olydis) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: please add
```
"x-ms-pageable": {
  "nextLinkName": null
}
```
(see other RPs), that will cause the `value` property of `OperationList` to be flattened out, i.e. the method's return type will be a collection rather than an object with one property that is a collection. Better UX 🙂

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (olydis) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: can those properties (and hence those of `Display`) be marked as readOnly?

**Review by olydis — COMMENTED** [review]
just some minor suggestions, very high quality specification and great documentation! 👍 🎉  note that my comments may also apply to the November one 🙂

File: specification/hanaonazure/resource-manager/readme.md
Comment (olydis) [inline specification/hanaonazure/resource-manager/readme.md]: `namespace`, `package-name` and `package-version` are programming language specific (C# could want the namespace "Azure.Management.SapHanaOnAzure"), so could you please move those 3 settings down into the `python` scope (e.g. after `payload-flattening-threshold`)? 🙂 

Just so we're not missing any potential quality issues: Did you see this in another RPs configuration?

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (olydis) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: can any property in here be marked as readOnly?
Also applies to `HardwareProfile`'s `hardwareType`, some `OSProfile` properties, `NetworkProfile` and `IpAddress`.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: Updated to just Operations_List

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: Added

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: Added

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: Since this is currently going to be a readOnly spec, I've made all the fields readOnly.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-06-15-preview/hanaonazure.json]: Yes, I've added them. Thanks!

**Comment by PakDLiu** [comment]
Is there anything else I need to do to get this merged in?

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/examples/HanaInstances_Get.json
Comment (marstr) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/examples/HanaInstances_Get.json]: Over the holidays, we updated the structure of this repository. Moving the `2017-11-03-preview` folder into a folder called `preview` would fix things. e.g.

`Microsoft.HanaOnAzure/2017-11-03-preview` -> `Microsoft.HanaOnAzure/preview/2017-11-03-preview`
 
For more information on what we did over the holiday, you can take a look at the wiki: 
[azures-rest-api-specs/wiki/December-2017-Refactoring](https://github.com/Azure/azure-rest-api-specs/wiki/December-2017-Refactoring)

**Review by marstr — CHANGES_REQUESTED** [review]

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/examples/HanaInstances_Get.json
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/2017-11-03-preview/examples/HanaInstances_Get.json]: Got it!

**Comment by PakDLiu** [comment]
I have updated the directory structure. Is there anything else I need to do to get this merged in?

**Comment by olydis** [comment]
@PakDLiu LGTM, @marstr are you happy too?

**Review by marstr — APPROVED** [review]

**Pull request merged**
